code/scripting/api/objs/mc_info.cpp

```diff
@@ -44,6 +44,25 @@ ADE_VIRTVAR(Model, l_ColInfo, "model", "The model this collision info is about",
 	return ade_set_args(L, "o", l_Model.Set(model_h(modelNum)));
 }
 
+ADE_FUNC(getCollisionSubmodel, l_ColInfo, nullptr, "The submodel where the collision occurred, if applicable", "submodel", "The submodel, or nil if none")
+{
+	mc_info_h *info;
+	submodel_h *sh = nullptr;
+
+	if (!ade_get_args(L, "o|o", l_ColInfo.GetPtr(&info), l_Submodel.GetPtr(&sh)))
+		return ade_set_error(L, "o", l_Submodel.Set(submodel_h()));
```

Comment:
The reason I did it this way was that, for better or worse, it appears most of the rest of the code does it that way.  Look at `getObjectFromSignature` or `getSoundentry` or `getStartDirectionInfo`.  In these cases (plus many others; search for "return ade_set_error") these return an invalid object rather than nil.

Reply:
In that case the documentation must be extended to make clear that an invalid request will give an invalid submodel and _not_ a nil for "no submodel found", which it currently implies